Update docs for prod environment setting (#3815)
fishtown-analytics/docs.getdbt.com: website/docs/guides/best-practices/environment-setup/2-one-deployment-environment.md

## What this looks like

1. You have a **single *development* environment** where dbt users can access the dbt Cloud IDE and make changes to their code on feature branches created off of your default branch in your repository (most often the `main` branch).

> 2. You have a **single *deployment* environment** (let’s call it “Production”) where your scheduled jobs run referencing the `main` branch. Be sure to set the environment as production to take advantage of features like dbt Explorer and cross-project refs. 

Reviewer: sugg making this a sub-bullet so it stands out more and doesn't get lost in the 2nd step

```suggestion
2. You have a **single *deployment* environment** (let’s call it “Production”) where your scheduled jobs run referencing the `main` branch. 
  * Make sure to set the environment to "Production" so you can take advantage of features like dbt Explorer and cross-project refs. 
```
